src/prefect/client/client.py

```diff
@@ -80,34 +80,158 @@ class Client:
     token will only be present in the current context.
 
     Args:
-        - api_server (str, optional): the URL to send all GraphQL requests
-            to; if not provided, will be pulled from `cloud.graphql` config var
+        - api_server (str, optional): the URL to send all GraphQL requests to; if not
+            provided, will be pulled from the current backend's `graphql` config
+            variable
+        - api_key (str, optional): a Prefect Cloud API key. If not provided, loaded
+            from `config.cloud.api_key` or from the on disk cache from the
+            `prefect auth` CLI
+        - tenant_id (str, optional): the Prefect tenant to use. If not provided, loaded
+            from `config.cloud.tenant_id` or the the on disk cache from the
+            `prefect auth` CLI
         - api_token (str, optional): a Prefect Cloud API token, taken from
             `config.cloud.auth_token` if not provided. If this token is USER-scoped, it may
             be used to log in to any tenant that the user is a member of. In that case,
             ephemeral JWTs will be loaded as necessary. Otherwise, the API token itself
-            will be used as authorization.
+            will be used as authorization. DEPRECATED; use `api_key` instead.
     """
 
-    def __init__(self, api_server: str = None, api_token: str = None):
+    def __init__(
+        self,
+        api_server: str = None,
+        api_key: str = None,
+        tenant_id: str = None,
+        api_token: str = None,
+    ):
         self._access_token = None
         self._refresh_token = None
         self._access_token_expires_at = pendulum.now()
-        self._active_tenant_id = None
+        self._tenant_id = None
         self._attached_headers = {}  # type: Dict[str, str]
         self.logger = create_diagnostic_logger("Diagnostics")
 
-        # store api server
-        self.api_server = api_server or prefect.context.config.cloud.get("graphql")
+        # Note the default is `cloud.api` which is `cloud.endpoint` or `server.endpoint`
+        # depending on the value of the `backend` key
+        self.api_server = api_server or prefect.context.config.cloud.api
```

Comment:
I made this change so that Agent clients and default clients would use the same api server but this breaks existing behavior and loading existing credentials -- I'll have to sort out a better way to do this :(

Reply:
(all the failing tests are from this change)

Reply:
I've fixed this by returning to the original default if an API key is not set